pubsub/azurepubsub/azurepubsub.go

```diff
@@ -301,47 +307,38 @@ type SubscriptionOptions struct {
 }
 
 // OpenSubscription initializes a pubsub Subscription on a given Service Bus Subscription and its parent Service Bus Topic.
-func OpenSubscription(ctx context.Context, parentNamespace *servicebus.Namespace, parentTopic *servicebus.Topic, sbSubscription *servicebus.Subscription, opts *SubscriptionOptions) *pubsub.Subscription {
-	ds := openSubscription(ctx, parentNamespace, parentTopic, sbSubscription, opts)
-	return pubsub.NewSubscription(ds, nil)
+func OpenSubscription(ctx context.Context, parentNamespace *servicebus.Namespace, parentTopic *servicebus.Topic, sbSubscription *servicebus.Subscription, opts *SubscriptionOptions) (*pubsub.Subscription, error) {
+	ds, err := openSubscription(ctx, parentNamespace, parentTopic, sbSubscription, opts)
+	if err != nil {
+		return nil, err
+	}
+	return pubsub.NewSubscription(ds, nil), nil
 }
 
 // openSubscription returns a driver.Subscription.
-func openSubscription(ctx context.Context, sbNs *servicebus.Namespace, sbTop *servicebus.Topic, sbSub *servicebus.Subscription, opts *SubscriptionOptions) driver.Subscription {
-	topicName := ""
-
-	if sbTop != nil {
-		topicName = sbTop.Name
+func openSubscription(ctx context.Context, sbNs *servicebus.Namespace, sbTop *servicebus.Topic, sbSub *servicebus.Subscription, opts *SubscriptionOptions) (driver.Subscription, error) {
+	if sbNs == nil {
```

Comment:
We really don't want to make a habit of these, or our code will be littered with them. Is there a particular reason you want to check here?

Reply:
Mostly because using the `URLOpener` makes it easier to accidentally omit arguments (e.g., by using `URLOpener`) and I think a more specific error message will make it easier to figure out when went wrong.